app/src/main/java/com/islamey/hamza/wazaker/ui/MainActivity.kt

```diff
@@ -1,49 +1,99 @@
+
+        setUpBottomViewNavigation()
+    }
+
+    private fun setDarkModeState() {
+        if (getDarkModeState(this))
+            AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_YES)
+        else
+            AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_NO)
     }
 
-    @SuppressLint("UseSwitchCompatOrMaterialCode")
-    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
+    private fun setUpBottomViewNavigation() {
+        val navHostFragment =
+            supportFragmentManager.findFragmentById(R.id.hostFragment) as NavHostFragment
+        navController = navHostFragment.navController
+
+        binding.bottomNavigationView.setOnItemSelectedListener { menuItem ->
+            when (menuItem.itemId) {
 
-        menuInflater.inflate(R.menu.main_menu, menu)
-        val itemSwitch: MenuItem = menu?.findItem(R.id.switch_action_bar)!!
-        itemSwitch.setActionView(R.layout.switch_layout)
+                R.id.home_action -> {
+                    navController.navigate(R.id.homeFragment)
+                    true
+                }
 
-        val switchButton =
-            menu.findItem(R.id.switch_action_bar).actionView?.findViewById(R.id.Main_switch) as SwitchButton
+                R.id.qibla_action -> {
+                    openQiblaActivity()
+                    true
+                }
 
-        switchButton.setOnCheckedChangeListener { _, isChecked ->
-            if (isChecked) AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_YES)
-            else AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_NO)
+                R.id.forty_action -> {
+                    navController.navigate(R.id.fortyHadithListFragment)
+                    true
+                }
 
+                R.id.settings_action -> {
+                    navController.navigate(R.id.settingsFragment)
+                    true
+                }
+
+                else -> false
+            }
         }
+    }
 
-        return true
+    private fun openQiblaActivity() {
+        val intent = Intent(this, CompassActivity::class.java)
+        intent.putExtra(Constants.TOOLBAR_BG_COLOR, "#FFFFFF") // Toolbar Background color
+        intent.putExtra(Constants.COMPASS_BG_COLOR, "#FFFFFF") // Compass background color
+        intent.putExtra(Constants.ANGLE_TEXT_COLOR, "#000000") // Angle Text color
+        intent.putExtra(Constants.DRAWABLE_DIAL, R.drawable.dial) // Your dial drawable resource
+        intent.putExtra(
+            Constants.DRAWABLE_QIBLA,
+            R.drawable.qibla
+        ) // Your qibla indicator drawable resource
+        intent.putExtra(
+            Constants.FOOTER_IMAGE_VISIBLE,
+            View.VISIBLE or View.INVISIBLE or View.GONE
+        ) // Footer World Image visibility
+        intent.putExtra(
+            Constants.LOCATION_TEXT_VISIBLE,
+            View.VISIBLE or View.INVISIBLE or View.GONE
+        ) // Location Text visibility
```

Comment:
is `CompassActivity` called from other places with different values ? 
if not so we do not need to pass these values.